cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -99,8 +99,14 @@ public final ProgramController run(final Program program, final ProgramOptions o
     final File cConfFile;
     final Program copiedProgram;
     final File programDir;    // Temp directory for unpacking the program
+    final String schedulerQueueName = options.getArguments().getOption(Constants.AppFabric.APP_SCHEDULER_QUEUE);
 
     try {
+
+      if (schedulerQueueName != null) {
+        hConf.set(Constants.MapReduce.MAP_REDUCE_JOB_QUEUE_NAME, schedulerQueueName);
```

Comment:
I thought about this, I did not want to override the run method and replicate the logic there. This way all logic for run is in AbstractDistributedProgramRunner